### PR TITLE
make supportedLifecycleRules configurable through the config

### PR DESCRIPTION
**Pull request by welansari**
This will help when unifying Cloudserver, as currently the value of the rules is a constant in Arsenal that is different between the 7.x and 8.x branches.

Issue: CLDSRV-588

### CODE REVIEW

**Review by benzekrimaha — APPROVED** [review]

File: tests/unit/Config.js [hunk 1]
```diff
@@ -668,4 +673,30 @@ describe('Config', () => {
             assert.throws(() => azureArchiveLocationConstraintAssert(locationObj));
         });
     });
+
+    describe('parseSupportedLifecycleRules', () => {
+        it('should assert when an empty array is provided', () => {
```
Comment (ghost) [inline tests/unit/Config.js]:
```suggestion
        it('should throw when an empty array is provided', () => {
```

File: tests/unit/Config.js [hunk 2]
```diff
@@ -668,4 +673,30 @@ describe('Config', () => {
...
+            assert.throws(() => parseSupportedLifecycleRules([]));
+        });
+
+        it('should use default values when no value is provided', () => {
+            const rules = parseSupportedLifecycleRules(undefined);
+            assert.deepStrictEqual(rules, [...supportedLifecycleRules]);
+        });
+
+        it('should assert when rule name is not valid', () => {
```
Comment (ghost) [inline tests/unit/Config.js]:
```suggestion
        it('should throw when rule name is not valid', () => {
```

**Review by ghost — APPROVED** [review]
+ maybe we can bump arsenal in this PR once the changes from https://github.com/scality/Arsenal/pull/2275 are merged

**Review by francoisferrand — APPROVED** [review]

**Comment by welansari** [comment]
> maybe we can bump arsenal in this PR once the changes from https://github.com/scality/Arsenal/pull/2275 are merged

See [here](https://github.com/scality/Arsenal/pull/2275#issuecomment-2553099231) for why i'm not doing it

**Pull request merged**